### PR TITLE
Feature: transient blueprint

### PR DESCRIPTION
**Pull request by iamyulong**
## Summary

Adds flag `is_transient` to `BlueprintDefinition`. 

This is to mark whether any component of a blueprint can be persisted or not

When set to `true`, objects of the blueprint will be prohibited from being persisted.

### For Internal Integrators

Updated `BlueprintDefinition` and `BlueprintDefinitionInit` to include the new flag.

Package publishing manifest now includes an additional parameter.

### CODE REVIEW

File: radix-engine/src/kernel/call_frame.rs [hunk 1]
```diff
@@ -376,33 +372,6 @@ impl<L: Clone> CallFrame<L> {
         &self.actor
     }
 
-    // TODO: Remove
-    fn get_type_info<S: SubstateStore>(
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:380]: YAY

File: radix-engine/src/kernel/call_frame.rs [hunk 2]
```diff
@@ -511,12 +480,18 @@ impl<L: Clone> CallFrame<L> {
         Ok((lock_handle, substate_value.len(), store_access))
     }
 
-    pub fn close_substate<S: SubstateStore>(
+    pub fn close_substate<S: SubstateStore, F, M>(
         &mut self,
         heap: &mut Heap,
         store: &mut S,
+        callback: &mut M,
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:487]: Can we remove callback object from this layer? I think the kernel itself should be able to add callback object on it's own.

File: radix-engine/src/kernel/call_frame.rs [hunk 3]
```diff
@@ -715,14 +690,20 @@ impl<L: Clone> CallFrame<L> {
...
-    pub fn create_node<'f, S: SubstateStore>(
+    pub fn create_node<'f, S: SubstateStore, F, M>(
         &mut self,
         node_id: NodeId,
         node_substates: NodeSubstates,
         heap: &mut Heap,
         store: &'f mut S,
-        push_to_store: bool,
-    ) -> Result<StoreAccessInfo, CreateNodeError> {
+        callback: &mut M,
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:699]: Ditto here

File: radix-engine/src/kernel/call_frame.rs [hunk 4]
```diff
@@ -824,15 +805,21 @@ impl<L: Clone> CallFrame<L> {
...
+    pub fn move_module<'f, S: SubstateStore, F, M>(
         &mut self,
         src_node_id: &NodeId,
         src_partition_number: PartitionNumber,
         dest_node_id: &NodeId,
         dest_partition_number: PartitionNumber,
         heap: &'f mut Heap,
         store: &'f mut S,
-    ) -> Result<StoreAccessInfo, MoveModuleError> {
+        callback: &mut M,
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:816]: Ditto here

File: radix-engine/src/kernel/call_frame.rs [hunk 5]
```diff
@@ -1069,15 +1056,21 @@ impl<L: Clone> CallFrame<L> {
         Ok((substates, store_access))
     }
 
-    pub fn drop_all_locks<S: SubstateStore>(
+    pub fn drop_all_locks<S: SubstateStore, F, M>(
         &mut self,
         heap: &mut Heap,
         store: &mut S,
-    ) -> Result<(), CloseSubstateError> {
+        callback: &mut M,
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:1063]: Ditto here

File: radix-engine/src/kernel/call_frame.rs [hunk 6]
```diff
@@ -1102,37 +1095,19 @@ impl<L: Clone> CallFrame<L> {
         self.owned_root_nodes.keys().cloned().collect()
     }
 
-    pub fn move_node_to_store<S: SubstateStore>(
+    pub fn move_node_to_store<S: SubstateStore, F, M>(
         heap: &mut Heap,
         store: &mut S,
+        callback: &mut M,
```
Comment (talekhinezh) [inline radix-engine/src/kernel/call_frame.rs:1101]: Ditto

File: radix-engine/src/kernel/call_frame.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/kernel/call_frame.rs:487]: I kind of need it inside the closure.

But we don't need `M` to be `KernelCallbackObject`. Maybe I just remove the type bound?

Another option is have trait object which encapsulates both the `callback` and the `filter`.

Comment (talekhinezh) [reply radix-engine/src/kernel/call_frame.rs:487]: Yes, exactly, the call_frame doesn't need to know that there is even such a thing as filter.

See a similar pattern here: https://github.com/radixdlt/radixdlt-scrypto/pull/1270/files#diff-d92ad4919aac15aa05fcea4887fe2e880ecd22d05a6b77dfd51479662372a17fL261

Comment (talekhinezh) [reply radix-engine/src/kernel/call_frame.rs:487]: right, that's what I meant

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**